### ISSUE TITLE
Suspecting bug in 'origins' handling

### ISSUE THREAD

**Issue by hinet2013** [walkor/phpsocket.io#225]
The foreach loop that handles the checks about the request's origins contains a forced return, so it always processes only the first entry of the array

The code that needs to be patched is:
https://github.com/walkor/phpsocket.io/blob/a80074f20c2e19b8cb3351ff718ae28e7df6bbea/src/Engine/Engine.php#L171

My correction proposal is the following:
```
if ($origin){

	$parts = parse_url($origin);
	$defaultPort = 'https:' === $parts['scheme'] ? 443 : 80;
	$parts['port'] = isset($parts['port']) ? $parts['port'] : $defaultPort;
	$allowed_origins = explode(' ', $this->origins);

	foreach( $allowed_origins as $allow_origin ){
		$ok =
				($allow_origin === $parts['scheme'] . '://' . $parts['host'] . ':' . $parts['port']) ||
				($allow_origin === $parts['scheme'] . '://' . $parts['host']) ||
				($allow_origin === $parts['scheme'] . '://' . $parts['host'] . ':*') ||
				($allow_origin === '*:' . $parts['port']);
			
		//OLD CODE: return call_user_func($fn, null, true, $req, $res);

		//NEW CODE
		if($ok){
			return call_user_func($fn, null, true, $req, $res);
		}
		//END NEW CODE
 	}
}
call_user_func($fn, null, false, $req, $res);
```

What do you think about?

**Comment by walkor** [comment]
Thank you for your post. This already fixed on 15 Feb 2019 by https://github.com/walkor/phpsocket.io/commit/e842dcd9150ea632801eec78f207f6651be5082c .

**Comment by hinet2013** [comment]
Composer is taking the last release v1.1.11 (that predates your correction)
Could the master branch considerable as stable or is better to apply a local patch (https://github.com/cweagans/composer-patches) while waiting for the next release?

Thank you

**Comment by walkor** [comment]
I released v1.1.12. Please upgrade.

**Comment by hinet2013** [comment]
Thank you very much, i appreciate it
